### PR TITLE
ipn/ipnlocal: add c2n handler to flush logtail for support debugging

### PR DESCRIPTION
**Pull request by bradfitz**
Updates tailscale/corp#8564

### CODE REVIEW

File: ipn/localapi/localapi.go [hunk 1]
```diff
@@ -914,6 +915,23 @@ func (h *Handler) serveFileTargets(w http.ResponseWriter, r *http.Request) {
...
+	if !h.PermitWrite {
+		http.Error(w, "access denied", http.StatusForbidden)
+		return
+	}
+	if r.Method != "POST" {
+		http.Error(w, "want POST", http.StatusMethodNotAllowed)
+		return
+	}
+	if h.b.TryFlushLogs() {
+		w.Header().Set("Content-Type", "application/json")
```
Comment (mihaip) [inline ipn/localapi/localapi.go:928]: `w.WriteHeader(http.StatusNoContent)` seems to be the more common way in which LocalAPI handlers indicate success with no body.

File: logtail/logtail.go [hunk 2]
```diff
@@ -461,12 +461,25 @@ func (l *Logger) upload(ctx context.Context, body []byte, origlen int) (uploaded
 	return true, nil
 }
 
-// Flush uploads all logs to the server.
-// It blocks until complete or there is an unrecoverable error.
+// Flush uploads all logs to the server. It blocks until complete or there is an
+// unrecoverable error.
+//
+// TODO(bradfitz): this apparently just returns nil. Not sure the history of it
```
Comment (mihaip) [inline logtail/logtail.go:467]: https://github.com/tailscale/corp/commit/9c2ec3565bb94f1d78de18988eb6b6b793209ed6 appears to be when it became a no-op

File: ipn/localapi/localapi.go [hunk 3]
```diff
@@ -73,6 +73,7 @@ var handler = map[string]localAPIHandler{
 	"dev-set-state-store":         (*Handler).serveDevSetStateStore,
 	"dial":                        (*Handler).serveDial,
 	"file-targets":                (*Handler).serveFileTargets,
+	"flush-logtail":               (*Handler).serveFlushLogtail,
```
Comment (mihaip) [inline ipn/localapi/localapi.go:76]: Do we need this LocalAPI method? As far as I can tell it's unused. Or were you thinking of add a `tailscale debug` command to also allow log flushing?

**Review by mihaip — APPROVED** [review]

File: ipn/localapi/localapi.go (hunk 1 above)
Comment (bradfitz) [reply ipn/localapi/localapi.go:928]: in this case I'm imagining including content here later, like some stats on what was buffered

File: ipn/localapi/localapi.go (hunk 3 above)
Comment (bradfitz) [reply ipn/localapi/localapi.go:76]: Yeah, this is mostly for debugging. But I'm also going to make "tailscale bugreport" etc do it.

I suppose I could remove this one and instead make this PR do it on bugreport.

Comment (bradfitz) [reply ipn/localapi/localapi.go]: Done

**Pull request merged**